Error if id_loan exists and has any duplicated value

maurolepore opened the pull request:
Closes #349

Thanks @georgeharris2deg

This PR makes `match_name()` fail if two things happen:

* The input loanbook has a column named `id_loan`.
* The column `id_loan` has at least one duplicated value.


@jdhoffa, notice that the loanbook may not have a column `id_loan` (should we enforce it?).

maurolepore commented on tests/testthat/_snaps/match_name.md:
```diff
@@ -195,3 +195,8 @@
       ]
     }
 
+# errors if any id_loan is duplicated (#349)
+
+      All values of `id_loan` in a `loanbook` must be unique.
+      Do you need to remove duplicated values (e.g. `1`)?
```
Here is what the users would see.

maurolepore commented on DESCRIPTION:
```diff
@@ -1,6 +1,6 @@
 Package: r2dii.match
 Title: Tools to Match Financial Portfolios with Climate Data
-Version: 0.0.8.9000
+Version: 0.0.8.9001
```
```r
remotes::install_github("2DegreesInvesting/r2dii.match@349_match-name-should-fail-earlier")
# or
remotes::install_github("2DegreesInvesting/r2dii.match@0.0.8.9001")
```

georgeharris2deg commented:
@maurolepore 
Thanks a lot for this 
If it isn't too late I would like to suggest editing the error message slightly 
**how it is now** 
"All values of `id_loan` in a `loanbook` must be unique.
      Do you need to remove duplicated values (e.g. `1`)?"
**Suggest edit** 
"All values of `id_loan` in a `loanbook` must be unique.
      Please ensure that every loan has a unique ID" 

**The reason being** - "remove Duplicate" could prompt a user to delete these rows. - These rows may have non-unique IDs however they may, in fact, unique loans. 
I.e. two loans to one company with the same ID.  (most likely due to an error on the bank's data entry side)
So we want to preserve both these loans and Identify them as being unique. 
i.e. two loans to one company with a unique ID each 

Thanks a lot !
@jdhoffa FYI

maurolepore commented:
Thanks @georgeharris2deg, I changed the error message a bit to more closely reflect what 
you suggest. I still would like to preserve the hint about the position of the first duplicated 
value (in the example below, the first duplicated value is `1`).

``` r
devtools::load_all()
#> Loading r2dii.match
duplicated <- rep.int(1, times = 2)
lbk <- fake_lbk(id_loan = duplicated)
ald <- fake_ald()

try(match_name(lbk, ald))
#> Error :   All values of `id_loan` in a `loanbook` must be unique (`1` is not).
#>   Please ensure that every loan has a unique identifier.
```

<sup>Created on 2021-02-11 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

--

@jdhoffa, I moved this to a draft because I see some snapshots are now out of date and I need to figure out what's going on.

maurolepore commented:
@jdhoffa, following the meeting we had last Friday I understand it's a priority to submit #353 to CRAN. I think it would be nice for that release to include this PR too -- as it was requested by @georgeharris2deg a while ago. 

I assume you don't have time to review so I'll go ahead and merge so I can start preparing the release. If you disagree, feel free to revert.

jdhoffa commented:
This looks good! Apologies for slacking on my review duties.
